src/types.h

```diff
@@ -459,15 +459,15 @@ class SizedType {
   }
   bool IsMapIterableTy() const
   {
-    return !IsMultiOutputMapTy();
+    return !IsMultiOutputMapTy() && (IsCastableMapTy() || !NeedsPercpuMap());
   }
 
   // These are special map value types that can't be reduced to a single value
-  // and output multiple lines when printed
+  // and output multiple lines when printed. In essence, once conceptual value
+  // written is actually multiple map keys.
   bool IsMultiOutputMapTy() const
```

Comment:
I think you've now changed the meaning of this function. `stats_t` IS a multi output map as it prints multiple things from whatever structure it keeps under the hood. I know you're trying to separate maps that keep multiple keys per value from ones that don't. In that case, I would say to rename this function to something like `IsMultiKeyMapTy` - just to remove the "output" part.

Reply:
You're right that there are ambiguous interpretations, but I don't think the previous definition holds either. Any aggregate type would then be a "multi output map", e.g.

```
struct foo { int x; int y; } BEGIN { @ = *(struct foo*)(0); print(@); }
```

But in this case `@` is not `IsMultiOutputMapTy`, despite being irreducible when printing.

I can change it per your suggestion, but not sure what the actual intent is here.

Reply:
> I disagree with that example. That prints 1 struct not multiple.

I don't understand. I'm saying that it's identical to `stats_t`. It is one struct, not multiple. You were saying that this did apply to `stats_t` because it is an aggregate type ("can't be reduced to a single value"), but this is true for all composite types and they were not considered "multi-output".

> The intent was just to be able to localize these odd map types that couldn't be used for things like casting, for loops, var assignment.

IMHO the problem is too many basic kinds and no way to build on top of them. Not trying to fix this here. But it's clear there's a problem when you need a category system on the top of them :)